tests/integration/test_reads.py

```diff
@@ -950,3 +951,11 @@ def test_read_from_s3_and_local_fs(catalog: Catalog, tmp_path: PosixPath) -> Non
 
     result_table = tbl.scan().to_arrow()
     assert result_table["colA"].to_pylist() == ["one", "one"]
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("catalog", [pytest.lazy_fixture("session_catalog_hive"), pytest.lazy_fixture("session_catalog")])
+def test_scan_with_datetime(catalog: Catalog) -> None:
+    table = create_table(catalog)
+    # test that this doesn't raise an exception...
+    table.scan(row_filter=GreaterThanOrEqual("datetime", datetime.now())).to_pandas()
```

Comment:
nit: the [table has `datetime` column](https://github.com/apache/iceberg-python/blob/8af4ca13e803122d4cd5a42517df5bcd06998ccf/tests/integration/test_reads.py#L68-L81), what if we add a row and then filter datetime before and after and assert the results. 

maybe something like,
```
yesterday = ...
table.append({"datatime": yesterday)
assert len(table.scan(row_filter=GreaterThanOrEqual("datetime", datetime.now()))) == 0
assert len(table.scan(row_filter=LessThanOrEqual("datetime", datetime.now()))) == 1
```

Reply:
thanks, looking much better now, took me too long to figure out I can run `make test-integration PYTEST_ARGS='-k test_scan_with_datetime'` to only select that test hahaha